### ISSUE TITLE
Ctrl-C on an empty command line in PyFerret does not get processed until the next command

### ISSUE THREAD

**Issue by karlmsmith** [NOAA-PMEL/Ferret#1373]
**Reported by karl smith on 7 Nov 2013 17:16 UTC**
If you enter Ctrl-C on an blank command line, the interrupted message and action is not processed until a command is entered.  Thus 
```
yes? (Ctrl-C) (Enter)
yes? (Enter)
yes? (Enter)
yes? say this (Enter)
this
** INTERRUPTED! **
yes? 
```

In Ferret, the readline call was configured to do various things (such as update X graphics and, I think, check the interrupt flag) while it was waiting on user input.  This is no longer the case (or possible) using Python readline for input.  If any command is given with the Ctrl-C, the interrupt is recognized when processing that command.  Thus the assignment of low priority.

I think all this needs is a check of the interrupt flag prior to each call prompting the user for input.

Migrated-From: http://dunkel.pmel.noaa.gov/trac/ferret/ticket/2101

**Comment by karlmsmith** [comment]
**Comment by karl.smith on 7 Nov 2013 18:20 UTC**
Actually, this behavior is the same in Ferret.  The only difference is in Ferret the Ctrl-C displays a "^C" when entered.

**Comment by karlmsmith** [comment]
**Comment by karl.smith on 13 Oct 2016 22:00 UTC**
In PyFerret the interrupted flag is cleared after each line is read from user typing at the command line.  This clears meaningless interrupted comment.
